### PR TITLE
Add support for non-pointer proto struct

### PR DESCRIPTION
**Pull request by alexshtin**
When Activity/Workflow parameter is a proto object, it had to be a pointer (because `proto.Message` is implemented on pointer receiver only), and now non-pointers are also supported (thanks to reflection magic):
```
ai := datapb.AccountInfo{...} // not &datapb.AccountInfo{...}
f := workflow.ExecuteActivity(ctx, "MakeTransaction", ai)
var tr datapb.TransactionResult // not *datapb.TransactionResult
f.Get(&tr)
```

### CODE REVIEW

File: converter/payload_converter_test.go [hunk 1]
```diff
@@ -160,19 +182,168 @@ func TestJsonPayloadConverter(t *testing.T) {
...
+	assert.Nil(t, payload)
+}
+
+func TestJsonPayloadConverter_ToPayload_Errors(t *testing.T) {
+	pc := NewJSONPayloadConverter()
+
+	var wt1 *testStruct
+	payload, err := pc.ToPayload(wt1)
+	require.NoError(t, err)
+	assert.Equal(t, "null", string(payload.Data))
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply converter/payload_converter_test.go:220]: I have it here too. This tests only JSON converter itself. But `NilPayloadConverter` is first in the chain: https://github.com/temporalio/sdk-go/blob/master/converter/default_data_converter.go#L29. So this shouldn't be a case with `DefaultDataConverter` but nothing blocks users to build custom chain of payload converters and then `nil` value can reach JSON converter.

File: converter/payload_converter_test.go [hunk 2]
```diff
@@ -160,19 +182,168 @@ func TestJsonPayloadConverter(t *testing.T) {
...
 }
 
-func TestProtoJsonPayloadConverter_NotPointer(t *testing.T) {
+func TestProtoJsonPayloadConverter_ToPayload_Errors(t *testing.T) {
+	pc := NewProtoJSONPayloadConverter()
+
+	var wt1 *commonpb.WorkflowType
+	_, err := pc.ToPayload(wt1)
+	require.Error(t, err)
+	assert.Equal(t, "unable to encode: Marshal called with nil", err.Error())
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply converter/payload_converter_test.go:200]: This is actually a good point. All 3 JSON encoders we use (Go JSON, google proto JSON, and gogo proto JSON) behaves differently on nil values. I would follow Go JSON convention which just returns `null`.

File: converter/payload_converter_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexshtin) [reply converter/payload_converter_test.go:245]: Added tests for `NilPayloadConverter` as it has some changes too.

**Review by mfateev — APPROVED** [review]

**Comment by mnichols** [comment]
In the future, if you need a google protobuf converter, here is what I was using in Cadence to support various versions of Google protobuf messages:
```
package workflows

import (
	"bytes"
	"fmt"
	"reflect"

	proto "github.com/golang/protobuf/proto" // nolint:staticcheck
	"go.uber.org/cadence/encoded"
	"go.uber.org/zap"
	protoV2 "google.golang.org/protobuf/proto"
	"weavelab.xyz/payments/pkg/messages"
	// nolint:staticcheck
)

type protobufDataConverter struct {
	logger *zap.Logger
}

func (dc *protobufDataConverter) ToData(value ...interface{}) ([]byte, error) {
	var data []byte
	buf := proto.NewBuffer(data)
	for _, v := range value {
		msg := messages.Message(v)
		if msg == nil {
			dc.logger.Warn(fmt.Sprintf("received non-protobuf message: %T", v))
		} else {
			err := buf.EncodeMessage(proto.MessageV1(msg))
			if err != nil {
				return nil, fmt.Errorf("error during encoding %T", v)
			}
		}
	}
	b := buf.Bytes()
	return b, nil
}
func (dc *protobufDataConverter) FromData(input []byte, valuePtr ...interface{}) error {
	if len(input) == 0 && len(valuePtr) > 0 {
		return fmt.Errorf("received no data for %d arguments", len(valuePtr))
	}
	buf := proto.NewBuffer(input)

	for _, v := range valuePtr {
		if msg, ok := v.(protoV2.Message); ok {
			if err := buf.DecodeMessage(proto.MessageV1(msg)); err != nil {
				return err
			}
		}
	}
	return nil
}
```

This comes with no warranty :) but maybe will help give a hand if the jump is made to Google protobufs.

**Pull request merged**